data-serving/data-service/test/model/date.test.ts

```diff
@@ -0,0 +1,39 @@
+import { Error } from 'mongoose';
+import { dateSchema } from '../../src/model/date';
+import mongoose from 'mongoose';
+
+/** A fake model with a field using the date schema. */
+const FakeModel = mongoose.model(
+    'FakeDocument',
+    new mongoose.Schema({
+        date: dateSchema,
+    }),
+);
+
+describe('validate', () => {
+    it('a type other than Date is invalid', async () => {
```

Comment:
Can we test (or document) what input will constitute a valid date? E.g., must be ISO-8601, or must be [...].

Reply:
sure. according to the documentation: "Mongoose will cast the value to a native JavaScript date using the Date() constructor." https://mongoosejs.com/docs/tutorials/dates.html

so basically any format that can be successfully cast to a date is OK -- which are these: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#Date_Time_String_Format

I've added a test for each valid format